### PR TITLE
fix: use target link libraries instead of ament target deps

### PR DESCRIPTION
**Pull request by wep21**

### CODE REVIEW

**Review by Ryanf55 — APPROVED** [review]
Perfect!

File: grid_map_cv/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ find_package(sensor_msgs REQUIRED)
 
 find_package(OpenCV REQUIRED
   COMPONENTS
-    opencv_photo
```
Comment (Ryanf55) [inline grid_map_cv/CMakeLists.txt:16]: Interesting... It seems both component names are supported and resolve. Is there a reason to switch?

Comment (wep21) [reply grid_map_cv/CMakeLists.txt:16]: Searching in github, components without `opencv_` prefix seem majority such as `core`, `highgui`, `improc`.

Comment (Ryanf55) [reply grid_map_cv/CMakeLists.txt:16]: I took a deep dive and found this is consistent with the intent of the config file that's exported.

**Pull request merged**